src/crepe/facade/SDLContext.h

```diff
@@ -1,48 +1,152 @@
 	SDLContext & operator=(SDLContext &&) = delete;
 
-	//TODO decide events wouter?
-
 private:
+	//! will only use handle_events
+	friend class LoopManager;
+	/**
+	* \brief Handles SDL events such as window close and input.
+	* \param running Reference to a boolean flag that controls the main loop.
+	*/
 	void handle_events(bool & running);
 
 private:
+	//! Will only use get_ticks
+	friend class AnimatorSystem;
+
+	/**
+	* \brief Gets the current SDL ticks since the program started.
+	* \return Current ticks in milliseconds as a constant uint64_t.
+	*/
+	const uint64_t get_ticks() const;
+
+private:
+	/**
+	* \brief Constructs an SDLContext instance.
+	* Initializes SDL, creates a window and renderer.
+	*/
 	SDLContext();
-	virtual ~SDLContext();
+
+	/**
+	* \brief Destroys the SDLContext instance.
+	* Cleans up SDL resources, including the window and renderer.
+	*/
+	~SDLContext();
 
 private:
+	//! Will use the funtions: texture_from_path, get_width,get_height.
 	friend class Texture;
-	SDL_Texture * texture_from_path(const char *);
-	//SDL_Texture* setTextureFromPath(const char*, SDL_Rect& clip, const int row, const int col);
+
+	//! Will use the funtions: texture_from_path, get_width,get_height.
+	friend class Animator;
+
+	/**
+	* \brief Loads a texture from a file path.
+    * \param path Path to the image file.
+    * \return Pointer to the created SDL_Texture.
+    */
+	SDL_Texture * texture_from_path(const std::string & path);
+
+	/**
+    * \brief Gets the width of a texture.
+    * \param texture Reference to the Texture object.
+    * \return Width of the texture as an integer.
+    */
+	int get_width(const Texture &) const;
+
+	/**
+    * \brief Gets the height of a texture.
+    * \param texture Reference to the Texture object.
+    * \return Height of the texture as an integer.
+    */
+	int get_height(const Texture &) const;
 
 private:
+	//! Will use draw,clear_screen, present_screen, camera.
 	friend class RenderSystem;
-	void draw(const Sprite &, const Transform &);
+
+	/**
+    * \brief Draws a sprite to the screen using the specified transform and camera.
+    * \param sprite Reference to the Sprite to draw.
+    * \param transform Reference to the Transform for positioning.
+    * \param camera Reference to the Camera for view adjustments.
+    */
+	void draw(const Sprite & sprite, const Transform & transform,
+			  const Camera & camera);
+
+	/**
+    * \brief Clears the screen, preparing for a new frame.
+         */
 	void clear_screen();
+
+	/**
+    * \brief Presents the rendered frame to the screen.
+    */
 	void present_screen();
 
+	/**
+	* \brief Sets the current camera for rendering.
+    * \param camera Reference to the Camera object.
+    */
+	void camera(const Camera & camera);
+
 private:
-	SDL_Window * game_window = nullptr;
-	SDL_Renderer * game_renderer = nullptr;
+	struct WindowDeleter {
+		void operator()(SDL_Window * window) { SDL_DestroyWindow(window); }
```

Comment:
Nu staat er een implementatie in de header. Dit kan toch gewoon als

SDLContext.h:
```cpp
std::unique_ptr<SDL_Window, std::function<void(SDL_Window *)>> game_window;
```

SDLContext.cpp:
```cpp
SDLWindow * window = SDL_CreateWindow(...);
if (!window)
  throw Exception(...);
this->game_window = {
  window,
  [](SDLWindow * window) { SDL_DestroyWindow(window); },
};
```